server/build_event_protocol/accumulator/accumulator.go

```diff
@@ -151,6 +157,12 @@ func (v *BEValues) populateWorkspaceInfoFromStructuredCommandLine(commandLine *c
 			switch environmentVariable {
 			case "CIRCLE_REPOSITORY_URL", "GITHUB_REPOSITORY", "BUILDKITE_REPO", "TRAVIS_REPO_SLUG", "GIT_URL", "CI_REPOSITORY_URL", "REPO_URL":
 				v.setStringValue(repoURLFieldName, value)
+			case "CIRCLE_BRANCH", "GITHUB_HEAD_REF", "BUILDKITE_BRANCH", "TRAVIS_BRANCH", "GIT_BRANCH", "CI_COMMIT_BRANCH":
+				v.setStringValue(branchNameFieldName, value)
+			case "GITHUB_REF":
+				if strings.HasPrefix(value, "refs/heads/") {
+					v.setStringValue(branchNameFieldName, strings.TrimPrefix(value, "refs/heads/"))
+				}
```

Comment:
"TrimPrefix returns s without the provided leading prefix string. If s doesn't start with prefix, s is returned unchanged."

so we could just do v.setStringValue(branchNameFieldName, strings.TrimPrefix(value, "refs/heads/"))